scripts/tests/test_release_archive.py

```
@@ -0,0 +1,276 @@
+def test_windows_targets_use_exe_suffix() -> None:
+    assert release_archive.binary_extension("x86_64-pc-windows-msvc") == ".exe"
+    assert release_archive.binary_extension("aarch64-apple-darwin") == ""
+
+
+def test_manifest_version_reports_missing_manifest(tmp_path: Path) -> None:
+    manifest = tmp_path / "Cargo.toml"
+
+    try:
+        release_archive.manifest_version(manifest)
+    except release_archive.ManifestVersionError as err:
+        assert err.manifest_path == manifest
```

Ruff (0.15.18)

[warning] Found assertion on exception `err` in `except` block, use `pytest.raises()` instead

(PT017)